From: Davide G. M. Salvetti <salve@debian.org>
Date: Tue, 21 Jun 2011 23:58:46 +0200
Subject: [PATCH] Fix preview with latest ghostscript.

- Backport of preview.el v1.285, in CVS since 2010-10-14.

Closes: #610714
Thanks: Faheem Mitha
---
 preview/preview.el |   31 +++++++++++++++++++------------
 1 files changed, 19 insertions(+), 12 deletions(-)

diff --git a/preview/preview.el b/preview/preview.el
index a2a17d3..e3658be 100644
--- a/preview/preview.el
+++ b/preview/preview.el
@@ -1,7 +1,7 @@
 ;;; preview.el --- embed preview LaTeX images in source buffer
 
-;; Copyright (C) 2001, 2002, 2003, 2004, 2005,
-;;               2006  Free Software Foundation, Inc.
+;; Copyright (C) 2001, 2002, 2003, 2004, 2005, 2006, 2010 Free
+;;   Software Foundation, Inc.
 
 ;; Author: David Kastrup
 ;; Keywords: tex, wp, convenience
@@ -23,7 +23,7 @@
 
 ;;; Commentary:
 
-;; $Id: preview.el,v 1.284 2009/06/18 19:20:46 angeli Exp $
+;; $Id: preview.el,v 1.285 2010/10/14 19:30:38 angeli Exp $
 ;;
 ;; This style is for the "seamless" embedding of generated images
 ;; into LaTeX source code.  Please see the README and INSTALL files
@@ -355,7 +355,7 @@ LIST consists of TeX dimensions in sp (1/65536 TeX point)."
   :group 'preview-gs
   :type 'string)
 
-(defcustom preview-gs-options '("-q" "-dSAFER" "-dNOPAUSE"
+(defcustom preview-gs-options '("-q" "-dDELAYSAFER" "-dNOPAUSE"
 				"-DNOPLATFONTS" "-dPrinted"
 				"-dTextAlphaBits=4"
 				"-dGraphicsAlphaBits=4")
@@ -1066,14 +1066,21 @@ NONREL is not NIL."
 
 (defun preview-prepare-fast-conversion ()
   "This fixes up all parameters for fast conversion."
-  (let ((file (if (consp (car preview-ps-file))
-		  (if (consp (caar preview-ps-file))
-		      (car (last (caar preview-ps-file)))
-		    (caar preview-ps-file))
-		(car preview-ps-file))))
+  (let* ((file (if (consp (car preview-ps-file))
+		   (if (consp (caar preview-ps-file))
+		       (car (last (caar preview-ps-file)))
+		     (caar preview-ps-file))
+		 (car preview-ps-file)))
+	 (all-files (if (and (consp (car preview-ps-file))
+			     (consp (caar preview-ps-file)))
+			(caar preview-ps-file)
+		      (list file))))
     (setq preview-gs-dsc (preview-dsc-parse file))
     (setq preview-gs-init-string
-	  (concat preview-gs-init-string
+	  (concat (format "{<</PermitFileReading[%s]>> setuserparams \
+.locksafe} stopped pop "
+			  (mapconcat 'preview-ps-quote-filename all-files ""))
+		  preview-gs-init-string
 		  (format "[%s(r)file]aload exch %s .runandhide aload pop "
 			  (preview-ps-quote-filename file)
 			  (preview-gs-dsc-cvx 0 preview-gs-dsc))))))
@@ -3512,7 +3519,7 @@ internal parameters, STR may be a log to insert into the current log."
 
 (defconst preview-version (eval-when-compile
   (let ((name "$Name: release_11_86 $")
-	(rev "$Revision: 1.284 $"))
+	(rev "$Revision: 1.285 $"))
     (or (when (string-match "\\`[$]Name: *release_\\([^ ]+\\) *[$]\\'" name)
 	  (setq name (match-string 1 name))
 	  (while (string-match "_" name)
@@ -3526,7 +3533,7 @@ If not a regular release, CVS revision of `preview.el'.")
 
 (defconst preview-release-date
   (eval-when-compile
-    (let ((date "$Date: 2009/06/18 19:20:46 $"))
+    (let ((date "$Date: 2010/10/14 19:30:38 $"))
       (string-match
        "\\`[$]Date: *\\([0-9]+\\)/\\([0-9]+\\)/\\([0-9]+\\)"
        date)
-- 
